rotkehlchen/exchanges/htx.py

```diff
@@ -188,24 +221,235 @@ def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
 
         return returned_balances, ''
 
+    def _paginated_query(
+            self,
+            endpoint: Literal['/v1/query/deposit-withdraw'],
+            options: dict[str, Any],
+            start_ts: Timestamp,
+    ) -> list[dict[str, Any]]:
+        """
+        Query endpoints that can return paginated data. At the moment this is used for asset
+        movements. For asset movements we use cursor pagination since for any time range
+        the api will return a valid response and we don't know when to stop querying until
+        we find a lower created-at than the start_ts(starting timestamp). The cursor pagination
+        uses the id of the last object in the last query to return the results of the next page.
+
+        May raise:
+        - RemoteError: if the network request fails
+        """
+        result = []
+        # copy since we are modifying the dict and could affect other queries using
+        # the same options
+        query_options = options.copy()
+        while len(output := self._query(absolute_path=endpoint, options=options)) != 0:
+            result.extend(output)
+            try:
+                query_options['from'] = output[-1]['id'] + 1  # from is inclusive, so we want after
+                created_at = output[-1]['created-at']  # the results are in descending order
+            except KeyError as e:
+                log.error(f'Missing expected keys {e!s} while iterating for HTX deposits and withdrawals')  # noqa: E501
+                break
+
+            if len(output) < PAGINATION_LIMIT or created_at < start_ts:
+                break
+
+        return result
+
+    def _query_deposits_withdrawals(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+            query_for: AssetMovementCategory,
+    ) -> list[AssetMovement]:
+        """
+        Process deposits/withdrawals from HTX. If any asset is unknown or we fail to
+        deserialize an entry the error is logged and we skip the entry.
+
+        This logic processes all entries returned by the API, discarding those that do not fall
+        within the range start_ts <= timestamp <= end_ts. Additionally, it does not terminate the
+        querying process until it encounters an created_at value that is lower than the start_ts.
+        """
+        log.debug(f'querying HTX online with {start_ts=}-{end_ts=}')
+        try:
+            raw_data = self._paginated_query(
+                endpoint='/v1/query/deposit-withdraw',
+                options={
+                    'type': 'deposit' if query_for == AssetMovementCategory.DEPOSIT else 'withdraw',  # noqa: E501
```

Comment:
1. Why do this and do one paginated query for deposits and one for withdrawals? Can't you just do one for all and then process the result?
2. Even if you did what you just did you should pass `deposit` or withdraw in the function to not have an if here